docs/upgrade.md

```diff
@@ -0,0 +1,42 @@
+# Upgrading KubeVirt
+
+This document outlines the step by step process of upgrading kubevirt
+from the perspective of the kubevirt-operator.  These steps can also
+be done by hand.
+
+**Major Upgrade** - Upgrade from 1.0 to 2.0
+**Minor Upgrade** - Upgrade from 1.0 to 1.1
+
+### Major Upgrade
+*kubevirt-1.1 to kubevirt-2.0*
+
+*When the kubevirt-operator sees the Virt CR change from version 1.1 to 2.0 it will...*
+ - Rollout kubevirt-2.0 virt-handler daemonsets
+ - Test virt-handler functionality with kubevirt-2.0 data
+ - Scale down virt-controllers pods
+ - Scale down virt-api pods
+ - Register kubevirt-2.0 CRD webhooks
+ - Update to kubevirt-2.0 CRDs
+ - Rollout kubevirt-2.0 virt-controllers
+ - Test virt-controller functionality with kubevirt-2.0 data
+ - Rollout kubevirt-2.0 virt-api
+ - Test virt-api functionality with kubevirt-2.0 data
+
+
+### Minor Upgrade
+*kubevirt-1.1 to kubevirt-1.2*
+
+*When the kubevirt-operator sees the Virt CR change from version 1.1 to 1.2 it will…*
+ - Rollout kubevirt-1.2 virt-handler daemonsets
+ - Test virt-handler functionality with kubevirt-1.1 data
+ - Rolling update of virt-controllers pods
+ - Test virt-controller functionality with kubevirt-1.1 data
+ - Rolling update of virt-api pods
+ - Test virt-api functionality with kubevirt-1.1 data
+ - Register kubevirt-1.2 CRD webhooks
+ - Update to kubevirt-1.2 CRDs
+ - Test virt-controller and virt-api functionality with kubevirt- 1.1 and kubevirt-1.2 data
+
+### Virt-launcher Upgrade
+The virt-launcher will be able to function with n-1 and n+1 releases.
```

Comment:
Let's specify this a little bit more. I think something like this is less ambiguous:

```
virt-launcher, which is part of release x.y will be compatible with:
 * all minor kubevirt versions >= y
 * at least the next major version (x + 1)
```

Let's also mention that we may introduce further restrictions once migrations are production ready. Something like

```
Once migrations are production-ready additional compatibility restrictions may be introduced. For example: virt-launcher from x.y can only migrate to other virt-launcher instances of a new major release x +1 and minor release 1. 
```